exchanges/deribit/deribit_websocket_eps.go

```diff
@@ -0,0 +1,2139 @@
+func (d *Deribit) WSRetriveCombos(currency string) ([]ComboDetail, error) {
+	if currency == "" {
+		return nil, fmt.Errorf("%w \"%s\"", errInvalidCurrency, currency)
+	}
+	var resp []ComboDetail
+	return resp, d.SendWSRequest(getCombos, map[string]string{"currency": currency}, &resp, false)
+}
+
+// WSCreateCombo verifies and creates a combo book or returns an existing combo matching given trades through the websocket connection.
+func (d *Deribit) WSCreateCombo(args []ComboParam) (*ComboDetail, error) {
+	if len(args) == 0 {
+		return nil, errNoArgumentPassed
+	}
+	for x := range args {
+		if args[x].InstrumentName == "" {
+			return nil, fmt.Errorf("%w, empty string", errInvalidInstrumentName)
+		}
+		args[x].Direction = strings.ToLower(args[x].Direction)
+		if args[x].Direction != sideBUY && args[x].Direction != sideSELL {
+			return nil, errors.New("invalid direction, only 'buy' or 'sell' are supported")
+		}
+		if args[x].Amount <= 0 {
+			return nil, errInvalidAmount
+		}
+	}
+	var resp ComboDetail
+	return &resp, d.SendWSRequest(createCombos, map[string]interface{}{"trades": args}, &resp, true)
+}
+
+// ------------------------------------------------------------------------------------------------
+
+// WSExecuteBlockTrade executes a block trade request
+// The whole request have to be exact the same as in private/verify_block_trade, only role field should be set appropriately - it basically means that both sides have to agree on the same timestamp, nonce, trades fields and server will assure that role field is different between sides (each party accepted own role).
+// Using the same timestamp and nonce by both sides in private/verify_block_trade assures that even if unintentionally both sides execute given block trade with valid counterparty_signature, the given block trade will be executed only once
+func (d *Deribit) WSExecuteBlockTrade(timestampMS time.Time, nonce, role, currency string, trades []BlockTradeParam) ([]BlockTradeResponse, error) {
+	if nonce == "" {
+		return nil, errMissingNonce
+	}
+	if role != "maker" && role != "taker" {
+		return nil, errInvalidTradeRole
+	}
+	if len(trades) == 0 {
+		return nil, errNoArgumentPassed
+	}
+	for x := range trades {
+		if trades[x].InstrumentName == "" {
+			return nil, fmt.Errorf("%w, empty string", errInvalidInstrumentName)
+		}
+		trades[x].Direction = strings.ToLower(trades[x].Direction)
+		if trades[x].Direction != sideBUY && trades[x].Direction != sideSELL {
+			return nil, errors.New("invalid direction, only 'buy' or 'sell' are supported")
+		}
+		if trades[x].Amount <= 0 {
+			return nil, errInvalidAmount
+		}
+		if trades[x].Price < 0 {
+			return nil, fmt.Errorf("%w, trade price can't be negative", errInvalidPrice)
+		}
+	}
+	signature, err := d.WSVerifyBlockTrade(timestampMS, nonce, role, currency, trades)
+	if err != nil {
+		return nil, err
+	}
+	if timestampMS.IsZero() {
+		return nil, errors.New("zero timestamp")
+	}
```

Comment:
You might be able to handle this in the WsVerifyBlockTrade above, or call it before the signature is called?